### PR TITLE
Fix invalid rspec version detection

### PR DESCRIPTION
**Pull request by ketiko**
Fixes grosser/rspec-instafail#20

### CODE REVIEW

File: lib/rspec/instafail.rb [hunk 1]
```diff
@@ -4,14 +4,14 @@ module RSpec
...
-    begin
+  if defined?(::RSpec::Core)
+    if ::RSpec::Core::Version::STRING >= '3.0.0'
+      require "rspec/instafail/rspec_3"
+    else
       require "rspec/instafail/rspec_2"
-    rescue LoadError # try rspec 1
-      require "rspec/instafail/rspec_1"
     end
+  elsif defined?(::Spec::VERSION)
```
Comment (grosser) [inline lib/rspec/instafail.rb]: not loading anything sounds wrong ... can this just be `else` ?

Comment (ketiko) [inline lib/rspec/instafail.rb]: If you do that then I belive you'll get an error when running rake tasks when using rspec 1.x, as it's the same issue.  Rspec isn't loaded and instalfail with raise an exception expecting rspec to be loaded.

We could try to require 'rspec/core' at the top, if it works, then test for the v3 version, catch the exception if 'rspec/core' doesn't load and then require 'spec' and load v1.

what do you think?

Comment (grosser) [inline lib/rspec/instafail.rb]: I'd rather leave the rspec1 code alone, if it worked before it will work now ... or did you have issues with rspec 1 ?

Comment (ketiko) [inline lib/rspec/instafail.rb]: I haven't had time to test rspec 1. I'll update the PR to be else rather than elsif so we load something by default.

**Comment by ketiko** [comment]
@grosser How about this?

**Comment by grosser** [comment]
looks good :)

**Comment by grosser** [comment]
0.3.0 is out!

**Pull request merged**